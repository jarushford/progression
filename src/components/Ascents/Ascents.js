import React from 'react'
import GradeSection from '../GradeSection/GradeSection'
import '../../main.scss'

export default function Ascents({ ascents }) {
  const gradeDivisions = ascents.reduce((grades, ascent) => {
    if (!grades[ascent.grade]) {
      grades[ascent.grade] = []
    }
    grades[ascent.grade].push(ascent)
    return grades
  }, {})
  const gradeKeys = Object.keys(gradeDivisions).sort((a, b) => {
    return b - a
  })
  const gradeRender = gradeKeys.map(grade => {
    return (
      <GradeSection
        grade={grade}
        ascents={gradeDivisions[grade]}
      />
    )
  })

  return (
    <section className="ascents">
      <div className="ascents-header">
        <h1 className="ascents-title">ASCENTS</h1>
        <button className="add-ascent">Add Ascent</button>
      </div>
      {gradeRender}
    </section>
  )
}