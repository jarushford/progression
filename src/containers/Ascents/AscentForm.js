import React, { Component } from 'react'

class AscentForm extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      location: '',
      caption: '',
      grade: '',
      ascentAdded: false
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    console.log(e.target.value)
    this.setState({ [name]: value })
  }

  handleSubmit = (e) => {
    e.preventDefault()

  }

  render() {
    const { name, location, caption, grade } = this.state
    return (
      <form className="ascent-form" onSubmit={this.handleSubmit}>
        <h2>ADD ASCENT</h2>
        <input name="name"
          value={name}
          onChange={this.handleChange}
          placeholder="name"
          autoFocus={true}
        />
        <input name="location"
          value={location}
          onChange={this.handleChange}
          placeholder="location"
        />
        <select id="grade"
          name="grade"
          value={grade}
          onChange={this.handleChange}
        >
          <option value="null"> -- </option>
          <option value="0"> V0 / 4 </option>
          <option value="1"> V1 / 5 </option>
          <option value="2"> V2 / 5+ </option>
          <option value="3"> V3 / 6A </option>
          <option value="4"> V4 / 6B </option>
          <option value="5"> V5 / 6C </option>
          <option value="6"> V6 / 7A </option>
          <option value="7"> V7 / 7A+ </option>
          <option value="8"> V8 / 7B+ </option>
          <option value="9"> V9 / 7C </option>
          <option value="10"> V10 / 7C+ </option>
          <option value="11"> V11 / 8A </option>
          <option value="12"> V12 / 8A+ </option>
          <option value="13"> V13 / 8B </option>
          <option value="14"> V14 / 8B+ </option>
          <option value="15"> V15 / 8C </option>
          <option value="16"> V16 / 8C+ </option>
        </select>
        <textarea name="caption"
          value={caption}
          onChange={this.handleChange}
          placeholder="caption"
        />
        <button>Add Ascent</button>
      </form>
    )
  }
}

export default AscentForm

