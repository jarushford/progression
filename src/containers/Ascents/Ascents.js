import React from 'react'
import GradeSection from '../GradeSection/GradeSection'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { uid } from 'react-uid'
import '../../main.scss'

export function Ascents({ ascents, user, disciplineBoulder }) {
  let gradeRender

  if (!user.name) {
    return <h1 className="no-user-msg">Log in or sign up to add ascents</h1>
  }

  if (!ascents.length) {
    gradeRender = <h1 className="no-data-msg">You have no saved ascents</h1>
  } else {
    const gradeDivisions = ascents.reduce((grades, ascent) => {
      if (!grades[ascent.grade]) {
        grades[ascent.grade] = []
      }
      grades[ascent.grade].push(ascent)
      return grades
    }, {})

    const allGradeKeys = Object.keys(gradeDivisions).sort((a, b) => b - a)
    let gradeKeys

    if (disciplineBoulder) {
      gradeKeys = allGradeKeys.filter(key => {
        return parseInt(key) < 17
      })
    } else {
      gradeKeys = allGradeKeys.filter(key => {
        return parseInt(key) > 17
      })
    }

    gradeRender = gradeKeys.map(grade => {
      return (
        <GradeSection
          grade={grade}
          ascents={gradeDivisions[grade]}
          key={uid(grade)}
        />
      )
    })
  }

  return (
    <section className="ascents">
      <div className="ascents-header">
        <h1 className="ascents-title">ASCENTS</h1>
        <Link to="/ascents/add">
          <button className="add-ascent">Add Ascent</button>
        </Link>
      </div>
      {gradeRender}
    </section>
  )
}

export const mapStateToProps = (state) => ({
  ascents: state.ascents,
  user: state.currentUser,
  disciplineBoulder: state.disciplineBoulder
})

export default connect(mapStateToProps)(Ascents)

Ascents.propTypes = {
  ascents: PropTypes.array.isRequired,
  user: PropTypes.object.isRequired,
  disciplineBoulder: PropTypes.bool.isRequired
}