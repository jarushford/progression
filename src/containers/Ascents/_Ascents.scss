.ascents-header {
  @include pageHeader();
}

.add-ascent {
  @include button();
}

.ascents-title {
  @include pageTitle();
}

.ascents {
  @include pageBody();
}

.grade-section {
  margin: 10px 18px 18px;

  h3 {
    color: $text;
    margin: 0;
    padding-bottom: 15px;
    border-bottom: 1px solid $ruleLine;
    font-weight: 600;
    font-size: 1.3rem;
  }
}

.ascent {
  border-bottom: 1px solid $ruleLine;

  p {
    color: $text;
    margin: 0 0 20px 7px;
  }
}

.ascent div {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.ascent-name {
  @include ascentDetails($accent, 1.7rem);
}

.ascent-location {
  @include ascentDetails($text, 1.1rem);
  margin-left: 20px;
  margin-bottom: 5px;
}

.ascent-form {
  @include form();

  input,
  button,
  select {
    @include form-element();
  }

  h2 {
    @include form-title();
  }

  textarea {
    @include form-caption()
  }

  button {
    @include button();
    width: 200px;
    margin: 30px auto;
  }
}

.content-section {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.fa-times {
  cursor: pointer;
  margin: 5px 10px 20px;
  font-size: 1.5rem;
  color: $text;
  transition: .4s ease;

  &:hover {
    font-size: 1.6rem;
    color: tomato;
  }
}

@media screen and (max-width: 450px) {
  .ascents-title {
    font-size: 1.4rem;
  }

  .add-ascent {
    width: 110px;
  }

  .ascents {
    width: 95%;
    margin: 160px auto;
  }

  .ascent-form {
    width: 95%;

    input,
    button,
    select,
    textarea {
      width: 250px;
      margin: 15px auto;
    }
  }
} 

@media screen and (max-width: 360px) {
  .ascent-form {
    width: 300px;
    margin-bottom: 20px;
    padding: 22px 26px;
  }
}