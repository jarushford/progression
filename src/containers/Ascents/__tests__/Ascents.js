import React from 'react'
import { shallow } from 'enzyme'
import { mapStateToProps, Ascents } from '../Ascents'

describe('Ascents', () => {
    let mockUser
    let mockAscents

    beforeEach(() => {
      mockUser = {
        name: 'Groot',
        email: 'groot@groot.groot',
        password: 'iamgroot',
        id: 3
      }
      mockAscents = [
        {
          name: 'Top Notch',
          location: 'RMNP, CO',
          grade: 'V13'
        },
        {
          name: 'Megalodon',
          location: 'Newlin, CO',
          grade: 'V7'
        },
        {
          name: 'Sabretooth',
          location: 'Thunder Ridge, CO',
          grade: 'V13'
        }
      ]
    })

  describe('Ascents Component', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<Ascents user={mockUser} ascents={mockAscents} disciplineBoulder={true} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with no user', () => {
      const noUser = { name: '', email: '', password: '' }
      const wrapper = shallow(<Ascents user={noUser}ascents={mockAscents} disciplineBoulder={true} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with no ascents', () => {
      const noAscents = []
      const wrapper = shallow(<Ascents user={mockUser} ascents={noAscents} disciplineBoulder={true} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should return a props object with data from store', () => {
      const mockState = {
        currentUser: mockUser,
        ascents: mockAscents
      }
      const expected = {
        user: mockUser,
        ascents: mockAscents
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
})