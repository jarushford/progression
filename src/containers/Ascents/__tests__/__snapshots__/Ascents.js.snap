// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ascents Ascents Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ascents
    ascents={
      Array [
        Object {
          "grade": "V13",
          "location": "RMNP, CO",
          "name": "Top Notch",
        },
        Object {
          "grade": "V7",
          "location": "Newlin, CO",
          "name": "Megalodon",
        },
        Object {
          "grade": "V13",
          "location": "Thunder Ridge, CO",
          "name": "Sabretooth",
        },
      ]
    }
    user={
      Object {
        "email": "groot@groot.groot",
        "id": 3,
        "name": "Groot",
        "password": "iamgroot",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ascents-header"
        >
          <h1
            className="ascents-title"
          >
            ASCENTS
          </h1>
          <Link
            replace={false}
            to="/ascents/add"
          >
            <button
              className="add-ascent"
            >
              Add Ascent
            </button>
          </Link>
        </div>,
        Array [
          <Connect(GradeSection)
            ascents={
              Array [
                Object {
                  "grade": "V13",
                  "location": "RMNP, CO",
                  "name": "Top Notch",
                },
                Object {
                  "grade": "V13",
                  "location": "Thunder Ridge, CO",
                  "name": "Sabretooth",
                },
              ]
            }
            grade="V13"
          />,
          <Connect(GradeSection)
            ascents={
              Array [
                Object {
                  "grade": "V7",
                  "location": "Newlin, CO",
                  "name": "Megalodon",
                },
              ]
            }
            grade="V7"
          />,
        ],
      ],
      "className": "ascents",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="ascents-title"
            >
              ASCENTS
            </h1>,
            <Link
              replace={false}
              to="/ascents/add"
            >
              <button
                className="add-ascent"
              >
                Add Ascent
              </button>
            </Link>,
          ],
          "className": "ascents-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ASCENTS",
              "className": "ascents-title",
            },
            "ref": null,
            "rendered": "ASCENTS",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="add-ascent"
              >
                Add Ascent
              </button>,
              "replace": false,
              "to": "/ascents/add",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Ascent",
                "className": "add-ascent",
              },
              "ref": null,
              "rendered": "Add Ascent",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "val-V13",
        "nodeType": "class",
        "props": Object {
          "ascents": Array [
            Object {
              "grade": "V13",
              "location": "RMNP, CO",
              "name": "Top Notch",
            },
            Object {
              "grade": "V13",
              "location": "Thunder Ridge, CO",
              "name": "Sabretooth",
            },
          ],
          "grade": "V13",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "val-V7",
        "nodeType": "class",
        "props": Object {
          "ascents": Array [
            Object {
              "grade": "V7",
              "location": "Newlin, CO",
              "name": "Megalodon",
            },
          ],
          "grade": "V7",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ascents-header"
          >
            <h1
              className="ascents-title"
            >
              ASCENTS
            </h1>
            <Link
              replace={false}
              to="/ascents/add"
            >
              <button
                className="add-ascent"
              >
                Add Ascent
              </button>
            </Link>
          </div>,
          Array [
            <Connect(GradeSection)
              ascents={
                Array [
                  Object {
                    "grade": "V13",
                    "location": "RMNP, CO",
                    "name": "Top Notch",
                  },
                  Object {
                    "grade": "V13",
                    "location": "Thunder Ridge, CO",
                    "name": "Sabretooth",
                  },
                ]
              }
              grade="V13"
            />,
            <Connect(GradeSection)
              ascents={
                Array [
                  Object {
                    "grade": "V7",
                    "location": "Newlin, CO",
                    "name": "Megalodon",
                  },
                ]
              }
              grade="V7"
            />,
          ],
        ],
        "className": "ascents",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="ascents-title"
              >
                ASCENTS
              </h1>,
              <Link
                replace={false}
                to="/ascents/add"
              >
                <button
                  className="add-ascent"
                >
                  Add Ascent
                </button>
              </Link>,
            ],
            "className": "ascents-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ASCENTS",
                "className": "ascents-title",
              },
              "ref": null,
              "rendered": "ASCENTS",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="add-ascent"
                >
                  Add Ascent
                </button>,
                "replace": false,
                "to": "/ascents/add",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Ascent",
                  "className": "add-ascent",
                },
                "ref": null,
                "rendered": "Add Ascent",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "val-V13",
          "nodeType": "class",
          "props": Object {
            "ascents": Array [
              Object {
                "grade": "V13",
                "location": "RMNP, CO",
                "name": "Top Notch",
              },
              Object {
                "grade": "V13",
                "location": "Thunder Ridge, CO",
                "name": "Sabretooth",
              },
            ],
            "grade": "V13",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "val-V7",
          "nodeType": "class",
          "props": Object {
            "ascents": Array [
              Object {
                "grade": "V7",
                "location": "Newlin, CO",
                "name": "Megalodon",
              },
            ],
            "grade": "V7",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Ascents Ascents Component should match the snapshot with no ascents 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ascents
    ascents={Array []}
    user={
      Object {
        "email": "groot@groot.groot",
        "id": 3,
        "name": "Groot",
        "password": "iamgroot",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ascents-header"
        >
          <h1
            className="ascents-title"
          >
            ASCENTS
          </h1>
          <Link
            replace={false}
            to="/ascents/add"
          >
            <button
              className="add-ascent"
            >
              Add Ascent
            </button>
          </Link>
        </div>,
        <h1
          className="no-data-msg"
        >
          You have no saved ascents
        </h1>,
      ],
      "className": "ascents",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="ascents-title"
            >
              ASCENTS
            </h1>,
            <Link
              replace={false}
              to="/ascents/add"
            >
              <button
                className="add-ascent"
              >
                Add Ascent
              </button>
            </Link>,
          ],
          "className": "ascents-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ASCENTS",
              "className": "ascents-title",
            },
            "ref": null,
            "rendered": "ASCENTS",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="add-ascent"
              >
                Add Ascent
              </button>,
              "replace": false,
              "to": "/ascents/add",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Ascent",
                "className": "add-ascent",
              },
              "ref": null,
              "rendered": "Add Ascent",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You have no saved ascents",
          "className": "no-data-msg",
        },
        "ref": null,
        "rendered": "You have no saved ascents",
        "type": "h1",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ascents-header"
          >
            <h1
              className="ascents-title"
            >
              ASCENTS
            </h1>
            <Link
              replace={false}
              to="/ascents/add"
            >
              <button
                className="add-ascent"
              >
                Add Ascent
              </button>
            </Link>
          </div>,
          <h1
            className="no-data-msg"
          >
            You have no saved ascents
          </h1>,
        ],
        "className": "ascents",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="ascents-title"
              >
                ASCENTS
              </h1>,
              <Link
                replace={false}
                to="/ascents/add"
              >
                <button
                  className="add-ascent"
                >
                  Add Ascent
                </button>
              </Link>,
            ],
            "className": "ascents-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ASCENTS",
                "className": "ascents-title",
              },
              "ref": null,
              "rendered": "ASCENTS",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="add-ascent"
                >
                  Add Ascent
                </button>,
                "replace": false,
                "to": "/ascents/add",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Ascent",
                  "className": "add-ascent",
                },
                "ref": null,
                "rendered": "Add Ascent",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You have no saved ascents",
            "className": "no-data-msg",
          },
          "ref": null,
          "rendered": "You have no saved ascents",
          "type": "h1",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Ascents Ascents Component should match the snapshot with no user 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ascents
    user={
      Object {
        "email": "",
        "name": "",
        "password": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Log in or sign up to add ascents",
      "className": "no-user-msg",
    },
    "ref": null,
    "rendered": "Log in or sign up to add ascents",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Log in or sign up to add ascents",
        "className": "no-user-msg",
      },
      "ref": null,
      "rendered": "Log in or sign up to add ascents",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
